skyportal/handlers/api/source.py

```diff
@@ -692,14 +692,20 @@ def post_source(data, user_id, session, refresh_source=True):
         raise AttributeError(
             "You must belong to one or more groups before you can add sources."
         )
-    try:
-        group_ids = [
-            int(id)
-            for id in data.pop('group_ids')
-            if int(id) in user_accessible_group_ids
-        ]
-    except KeyError:
+
+    group_ids = data.pop('group_ids')
+    if group_ids is None:
         group_ids = user_group_ids
+    else:
+        group_ids_tmp = []
```

Comment:
Thanks for fixing that! Though I feel like you could even have made it happen without any extra list. All that this does is validate that the ids are accessible to the user and then make sure they are ints. You could have also just made them ints with a one liner and then loop over them and raise the error when one of them was invalid.

But that aside, this seems to work just fine! 

LGTM. Merge once green.